### ISSUE TITLE
508-defect-2 [COGNITION, SCREEN READER]: Screen reader users SHOULD know there is relevant info after making a selection in the wizard

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#16745]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** Josh Kim

## User Story or Problem Statement

As a screen reader user, I need to know that there is additional relevant information after specific radio button selections. Since I may be accustomed to navigating forms using `tab` , I may accidentally skip relevant information in search of a `submit` button.

## Details

Two undesired scenarios exist as a screen reader user navigates the wizard:
- If a selection dynamically displays a box with a paragraph of information and a link, the user may `tab` directly to the link without being able to gather context as to why that link is needed.
- If a selection dynamically displays a box with a paragraph of information and _no_ link, the user may `tab` directly out of the form itself. 

## Acceptance Criteria

- [ ] All radio selections that dynamically load information should be linked to that information using `aria-describedby`

## Environment

* Operating System: Mac, Windows
* Browser: Safari, Firefox
* Screenreading device: VoiceOver, NVDA
* Server destination: staging, production

## Steps to Recreate

1. [Open the wizard for form 28-8832](https://staging.va.gov/careers-employment/education-and-career-counseling/apply-career-guidance-form-28-8832/introduction).
2. Start any screenreading device. 
3. Tab to the form and select "Veteran."
4. Tab and select "No."
5. Tab and select "No."
6. Tab and select "No." You should hear no additional information.
7. Tab, which jumps to the Need Help number.

## Solution (if known)
A more optimally accessible solution would be to split the wizard into separate pages [(see gov.uk's check eligibility example)](https://design-system.service.gov.uk/patterns/check-a-service-is-suitable/) as opposed to having additional content dynamically loaded. This would mitigate a substantial amount of known unknowns in terms of usability and accessibility errors that come with the wizard pattern.

If that's not possible, you could consider the following solution below for every instance where a radio selection generates informational content. This will announce the information upon selection of the triggering radio input by using `aria-describedby` which programmatically ties the information to the radio input.

### Before

```html
<!-- Radio input that generates information -->
<input id="discharge-status-1" name="discharge-status" type="radio" value="dischargeNotification">

<!-- Information that is generated by the radio input -->
<div class="vads-u-margin-top--2 vads-u-padding--3 vads-u-background-color--gray-lightest">
  <p class="vads-u-margin--0">
    To be eligible for career planning and guidance, you or your sponsor must have been discharged from active duty in the last year, or have less than 6 months until discharge.
  </p>
</div>
```

### After
```html
<!-- Radio input that generates information now has aria-describedby -->
<input id="discharge-status-1" name="discharge-status" type="radio" value="dischargeNotification" aria-describedby="unique-id-1">

<!-- Information that is generated by the radio input now has a unique id -->
<div id="unique-id-1" class="vads-u-margin-top--2 vads-u-padding--3 vads-u-background-color--gray-lightest">
  <p class="vads-u-margin--0">
    To be eligible for career planning and guidance, you or your sponsor must have been discharged from active duty in the last year, or have less than 6 months until discharge.
  </p>
</div>
```

## WCAG or Vendor Guidance (optional)

* [WCAG 2.1 Level A - 1.3.2 Meaningful Sequence](https://www.wuhcag.com/meaningful-sequence/) - Users who rely on assistive technology (such as a screen reader) to interpret content, require content to be presented in a meaningful order. If this is presented out of sequence, users may become disorientated and will not understand the content.
* [Gov.uk's check a service is eligible pattern](https://design-system.service.gov.uk/patterns/check-a-service-is-suitable/) would be a significantly more accessible and usable pattern to address this need. Until more testing is conducted with screen readers, the wizard pattern should be used cautiously.

## Screenshots or Trace Logs
Upon selecting "no", the screen reader should read out the information using `aria-describedby` 
<img width="727" alt="Screen Shot 2020-12-01 at 1 43 49 PM" src="https://user-images.githubusercontent.com/14154792/100783488-d63b5880-33db-11eb-898f-bd03c008560d.png">

**Comment by 1Copenut** [comment]
@joshkimux Let's rally on this one. I wrote an issue ticket some time back but didn't get very far: https://github.com/department-of-veterans-affairs/va.gov-team/issues/316

I've let the Design System team know about it as well. Together we can come up with a good solution.

**Comment by joshkimux** [comment]
I believe there should be an option to use `aria-describedby` in the wizard now!
https://github.com/department-of-veterans-affairs/va.gov-team/issues/316

**Comment by tblackwe** [comment]
Assigned to @DLarson-Oddball

**Comment by DLarson-Oddball** [comment]
@tblackwe Reviewed this ticket and the functionality works as expected. When I use a screen reader and select no, the text in the paragraph below the option is read. This behavior holds true for all other options.

**Comment by tblackwe** [comment]
TY @DLarson-Oddball, closing this one out as completed.
